src/common/proc_macro/src/session_config.rs

```diff
@@ -0,0 +1,262 @@
+// Copyright 2023 RisingWave Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use bae::FromAttributes;
+use proc_macro2::TokenStream;
+use proc_macro_error::{abort, OptionExt, ResultExt};
+use quote::{format_ident, quote, quote_spanned};
+use syn::DeriveInput;
+
+#[derive(FromAttributes)]
+pub struct Parameter {
```

Comment:
May add some docs and doctest to demonstrate usage

Reply:
Oh it's on `session_config`. 

```suggestion
/// See [`session_config`]
```

Reply:
It's actually no need to pub